lib/sensors/src/temperature.rs

```diff
@@ -0,0 +1,122 @@
+use hyped_io::i2c::HypedI2c;
+
+/// Temperature implements the logic to read the temperature from the STTS22H temperature sensor
+/// using the I2C peripheral provided by the HypedI2c trait. The temperature sensor is configured
+/// to continuous mode with a sampling rate of 200Hz. The temperature is read from the sensor and
+/// converted to a floating point value in degrees Celsius.
+///
+/// Data sheet: https://www.st.com/resource/en/datasheet/stts22h.pdf
+pub struct Temperature<T: HypedI2c> {
+    i2c: T,
+    device_address: u8,
+}
+
+impl<T: HypedI2c> Temperature<T> {
+    /// Create a new instance of the temperature sensor and attempt to configure it
+    pub fn new(mut i2c: T) -> Result<Self, ()> {
```

Comment:
probably take in a `TemperatureAddresses` value as an argument cause this should be generic over all 4 addresses

Reply:
Done :)